src/reducers/copy.js

```diff
@@ -0,0 +1,98 @@
+/**
+*
+* @licstart  The following is the entire license notice for the JavaScript code in this file.
+*
+* Merge MARC records
+*
+* Copyright (C) 2015-2019 University Of Helsinki (The National Library Of Finland)
+*
+* This file is part of marc-record-merge-js
+
+* marc-record-merge-js program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* marc-record-merge-js is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+
+// Hakee basesta ja sourcesta copy.spec.js:ssä määritellyn patternin (patternTest) mukaiset kentät
+export default (pattern) => (base, source) => {
+  const baseFields = base.get(pattern);
+  // eslint-disable-next-line no-console
+  console.log('baseFields: ');
+  // eslint-disable-next-line no-console
+  console.log(baseFields);
+  const sourceFields = source.get(pattern);
+  // eslint-disable-next-line no-console
+  console.log('sourceFields: ');
+  // eslint-disable-next-line no-console
+  console.log(sourceFields);
+  return copyFields();
+
+  // CopyFields tässä on nimellä createReducer copy.spec.js:ssä
+  // Kaikki toiminnallisuus menee copyFieldsin sisään
+  function copyFields() {
+    // Testi 01: Jos basessa ei ole kenttää lainkaan, se kopioidaan sourcesta baseen
+    // Testissä 01 basessa ei ole lainkaan kenttää 010 (pattern) joten se kopioidaan sourcesta.
+    // InsertField tulee MarcRecordista
+    if (baseFields.length === 0) {
+      sourceFields.forEach(f => base.insertField(f));
+      // eslint-disable-next-line no-console
+      console.log('base after test 01: ');
+      // eslint-disable-next-line no-console
+      console.log(base);
+      return base;
+    }
+    // Jos basessa on jo kenttä, sourcen kenttä kopioidaan uutena basen kentän lisäksi
+    // Eli merged.jsonissa voi olla useampi kappale samaa kenttää.
+
+    // MissingFieldsillä etsitään mitkä kentät puuttuvat
+    const missingFields = sourceFields.filter(sourceField => {
```

Comment:
Since the callback function is long, don't implement inline but at the end of the enclosing function instead. This then becomes, for example: `const missingFields = sourceFields.filter(filterMissing);`